components/org.wso2.carbon.metrics.impl/src/test/java/org/wso2/carbon/metrics/impl/MetricApiTest.java

```diff
@@ -128,4 +127,35 @@ public void testGetMetricWithDifferentLevel() {
             Assert.assertThat(e, IsInstanceOf.instanceOf(IllegalArgumentException.class));
         }
     }
+
+    public void testDeleteMetrics() {
+        try {
+            MetricManager.meter("org.wso2.main.throughput", Level.INFO);
+            assertTrue("Error occurred while deleting the metric from the registry",
+                    metricService.removeMetric("org.wso2.main.throughput"));
+        } catch (Exception e) {
+            fail("Metric should be removed");
+        }
+    }
+
+    public void testDeleteMetricsWithAnnotatedName() {
+        try {
+            MetricManager.meter("org.wso2.main[+].sub.throughput", Level.INFO);
+            metricService.removeMetric("org.wso2.main.throughput");
+            fail("Should throw an exception, cannot delete metric when there's no sufficient Levels"
```

Comment:
This is a wrong test case, right? We must test whether metrics created by annotated name are removed.